libc/include/llvm-libc-macros/limits-macros.h

```diff
@@ -0,0 +1,252 @@
+//===-- Definition of macros from limits.h --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __LLVM_LIBC_MACROS_LIMITS_MACROS_H
+#define __LLVM_LIBC_MACROS_LIMITS_MACROS_H
+
+#if __has_include_next(<limits.h>)
+
+// Suppress `#include_next is a language extension` warnings.
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wgnu-include-next"
+#else // gcc
+#pragma GCC system_header
+#endif // __clang__, __GNUC__
+
+#if defined(__GNUC__) && !defined(__clang__) && !defined(_GCC_LIMITS_H_)
+// The system's limits.h may, in turn, try to #include_next GCC's limits.h when
+// building with GCC.
+// Define the macro _GCC_LIMITS_H_ to stop its chains of #include_next.
+#define _GCC_LIMITS_H_
+#endif
+
+// Include compiler's header
```

Comment:
Can you confirm that this is the compiler's header?

I guess I would have thought that `#include <limits.h>` _is_ the compiler's header, so `#include_next <limits.h>` is...perhaps `/usr/include/limits.h` which would be from glibc. The `-H` compiler flag can give you a very verbose output about the chain of includes.

Reply:
Drop `#include_next <limits.h>` to get out of these messy chains of header inclusion.